Add fuzzing audit blog post (#5827)
open-telemetry/opentelemetry.io: content/en/blog/2024/fuzzing-audit-results.md

---
title: Opentelemetry Collector Completes Fuzzing Audit
linkTitle: Fuzzing Audit Results

> date: 2024-12-19

Reviewer: @AdamKorcz @open-telemetry/docs-approvers do we want to publish this today or move it early next year?

Author: I say we publish this week.

author: '[Adam Korczynski](https://github.com/AdamKorcz)'

> issue:

Reviewer: ```suggestion
issue: 5798
```

sig: GC
---

Opentelemetry is happy to announce the completion of the Collectors fuzzing audit sponsored by [the CNCF](https://www.cncf.io/) and carried out by [Ada Logics](https://adalogics.com/). The audit marks a significant step in the Opentelemetry project ensuring the security and reliability of the collector for its users. 

Fuzzing is a testing technique that involves executing an API with a high amount of pseudo random inputs and observing the APIs behaviour. The technique has increased in popularity due to its empirical success in finding security vulnerabilities and reliability issues. Fuzzing initially developed with a focus on testing software implemented in memory-unsafe languages for which fuzzing has been most productive. However, in recent years, fuzzing has expanded to memory-safe languages alike for which the outcome has also been rewarding. 

> Over several years, the CNCF has invested in fuzzing for its ecosystem which has resulted in finding numerous security vulnerabilities in widely used projects such as Helm (CVE-2022-36055, CVE-2022-23524, CVE-2022-23526, CVE-2022-23525), the Notary project (CVE-2023-25656), containerd (CVE-2023-25153), Crossplane (CVE-2023-28494, CVE-2023-27483) and Flux (CVE-2022-36049). 

Reviewer: ```suggestion
Over several years, the CNCF has invested in fuzzing for its ecosystem. This testing has found numerous security vulnerabilities in widely used projects such as [Helm](https://helm.sh/) (CVE-2022-36055, CVE-2022-23524, CVE-2022-23526, CVE-2022-23525), the [Notary Project](https://notaryproject.dev/) (CVE-2023-25656), [containerd](https://containerd.io/) (CVE-2023-25153), [Crossplane](https://www.crossplane.io/) (CVE-2023-28494, CVE-2023-27483) and [Flux](https://fluxcd.io/) (CVE-2022-36049). 
```

> A summary of the audit has been published [here](https://github.com/open-telemetry/community/blob/main/reports/ADA_Logics-collector-fuzzing-audit-2024.pdf).

Reviewer 2: Could we add a quick recap of the results so far? Were any crashes detected as of now? I couldn't see any in the audit summary, but if that's wrong, please correct me!

```suggestion
## The results so far

Fuzz testing for the Collector is ongoing, allowing for changes to the project to be tested as well. As of the date of this post, no crashes have been detected.

But the work is not done! The Ada Logics team created the Collector's fuzzing setup as a reference implementation that other OpenTelemetry subprojects can rely on to create their own fuzz testing, ensuring greater stability for the project as a whole.

For more insight into the audit process, see the [published summary](https://github.com/open-telemetry/community/blob/main/reports/ADA_Logics-collector-fuzzing-audit-2024.pdf).
```

---
title: OpenTelemetry Collector Completes Fuzzing Audit
linkTitle: Fuzzing Audit Results

> date: 2024-12-19

Reviewer: ```suggestion
date: 2024-12-20
```

---
title: OpenTelemetry Collector Completes Fuzzing Audit
linkTitle: Fuzzing Audit Results
date: 2024-12-20
author: '[Adam Korczynski](https://github.com/AdamKorcz)'
issue: 5798

> sig: GC

Reviewer: ```suggestion
sig: GC
cSpell:ignore: Korczynski containerd
```

---
title: OpenTelemetry Collector Completes Fuzzing Audit
linkTitle: Fuzzing Audit Results
date: 2024-12-20
author: '[Adam Korczynski](https://github.com/AdamKorcz)'
issue: 5798
sig: GC
cSpell:ignore: Korczynski containerd
---

OpenTelemetry is happy to announce the completion of the Collector's fuzzing

> audit sponsored by [the CNCF](https://www.cncf.io/) and carried out by

Reviewer 3: Catching up on this: @AdamKorcz - do you have a link that supports the "CNCF sponsored this" claim?

Author: @chalin Please send me an email at adam@adalogics.com and I can put you in touch with someone in the CNCF who can confirm this.
